### PR TITLE
Add VPG #16

### PR DESCRIPTION
**Pull request by Het-Shah**
Reused the PPO code. Modified the loss and policy and value function's update rules.

### CODE REVIEW

File: jigglypuffRL/classicalrl/vpg/vpg.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        # compute policy and value loss
+        clipping = (
+            torch.clamp(self.policy_fn.policy_hist, 1 - self.clip_param, 1 + self.clip_param)
+            .mul(A)
+            .to(self.device)
+        )
+
+        loss_policy = (
+            torch.mean(torch.min(torch.mul(self.policy_fn.policy_hist, A), clipping)).mul(-1).unsqueeze(0)
+        )
```
Comment (ajaysub110) [inline jigglypuffRL/classicalrl/vpg/vpg.py:142]: The policy loss calculation appears to be incorrect. In standard VPG, there is no clipping of policies. Rather, the loss should simply be:
```python
loss_policy = torch.sum(
  torch.mul(self.policy_fn.policy_hist, A)
).mul(-1).unsqueeze(0)
```
Note that I've replaced mean with sum since VPG averages loss only with respect to the number of trajectories. (Refer https://spinningup.openai.com/en/latest/algorithms/vpg.html)

**Comment by Sharad24** [comment]
If its really using a `MlpPolicy` maybe we should just have it in `deeprl`

**Comment by ajaysub110** [comment]
Agree. It also uses `MlpValue` for advantage estimation

**Comment by ajaysub110** [comment]
@Het-Shah can you move this to `deeprl`?

**Comment by ajaysub110** [comment]
And please create branches for your PRs from next time since there may be merge conflicts

**Comment by Het-Shah** [comment]
OK will do the changes. Sorry for the errors.
